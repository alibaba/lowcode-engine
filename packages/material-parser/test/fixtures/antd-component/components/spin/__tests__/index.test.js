import React from 'react';
import { render, mount } from 'enzyme';
import Spin from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Spin', () => {
  mountTest(Spin);
  rtlTest(Spin);

  it('should only affect the spin element when set style to a nested <Spin>xx</Spin>', () => {
    const wrapper = mount(
      <Spin style={{ background: 'red' }}>
        <div>content</div>
      </Spin>,
    );
    expect(
      wrapper
        .find('.ant-spin-nested-loading')
        .at(0)
        .prop('style'),
    ).toBeFalsy();
    expect(
      wrapper
        .find('.ant-spin')
        .at(0)
        .prop('style').background,
    ).toBe('red');
  });

  it("should render custom indicator when it's set", () => {
    const customIndicator = <div className="custom-indicator" />;
    const wrapper = render(<Spin indicator={customIndicator} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be controlled by spinning', () => {
    const wrapper = mount(<Spin spinning={false} />);
    expect(wrapper.instance().state.spinning).toBe(false);
    wrapper.setProps({ spinning: true });
    expect(wrapper.instance().state.spinning).toBe(true);
  });

  it('if indicator set null should not be render default indicator', () => {
    const wrapper = mount(<Spin indicator={null} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should support static method Spin.setDefaultIndicator', () => {
    Spin.setDefaultIndicator(<em className="custom-spinner" />);
    const wrapper = mount(<Spin />);
    expect(wrapper).toMatchSnapshot();
    Spin.setDefaultIndicator(null);
  });
});
