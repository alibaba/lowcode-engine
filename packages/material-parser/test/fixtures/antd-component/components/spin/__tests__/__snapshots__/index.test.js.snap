// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin if indicator set null should not be render default indicator 1`] = `
<Spin
  indicator={null}
  size="default"
  spinning={true}
  wrapperClassName=""
>
  <div
    className="ant-spin ant-spin-spinning"
  />
</Spin>
`;

exports[`Spin rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-spin ant-spin-spinning ant-spin-rtl"
>
  <span
    class="ant-spin-dot ant-spin-dot-spin"
  >
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
    <i
      class="ant-spin-dot-item"
    />
  </span>
</div>
`;

exports[`Spin should render custom indicator when it's set 1`] = `
<div
  class="ant-spin ant-spin-spinning"
>
  <div
    class="custom-indicator ant-spin-dot"
  />
</div>
`;

exports[`Spin should support static method Spin.setDefaultIndicator 1`] = `
<Spin
  size="default"
  spinning={true}
  wrapperClassName=""
>
  <div
    className="ant-spin ant-spin-spinning"
  >
    <em
      className="custom-spinner ant-spin-dot"
    />
  </div>
</Spin>
`;
